packages/desktop-client/src/components/filters/FiltersMenu.js

```diff
@@ -338,22 +339,45 @@ function ConfigureField({
 
 function ButtonType({ type, dispatch }) {
   return (
-    <Button
-      type="bare"
-      onClick={() => dispatch({ type: 'select-field' })}
-      title={type && 'Filters'}
-    >
+    <View>
       {type === 'reports' ? (
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏uhh.. this is why I really did not like the introduction of the `type` prop here. We're now cramming more logic into here. Eventually it will just grow into unmaintainable mess.

Perhaps time to take a step back and restructure it so we could compose using it? Or even extract it into separate components instead of cramming stuff together?

Reply:
Okay I simplified it and borke it into it's own file. LMK!